framework/src/modules/pos/module.ts

```diff
@@ -755,7 +756,16 @@ export class PoSModule extends BaseModule {
 			const numInitValidators =
 				genesisData.initRounds + this._moduleConfig.numberActiveValidators - round;
 			const numElectedValidators = this._moduleConfig.numberActiveValidators - numInitValidators;
-			const activeValidators = snapshotValidators.slice(0, numElectedValidators);
+			let weightSum = BigInt(0);
+			const activeValidators = snapshotValidators.slice(0, numElectedValidators).map(v => {
+				const scaledWeight = this._ceiling(v.weight, WEIGHT_SCALE_FACTOR);
+				weightSum += scaledWeight;
+				return {
+					...v,
+					weight: scaledWeight,
+				};
+			});
+			const averageWeight = weightSum / BigInt(numElectedValidators);
 			for (const address of genesisData.initValidators) {
 				// when activeValidator is filled, don't add anymore
 				if (activeValidators.length === this._moduleConfig.numberActiveValidators) {
```

Comment:
this can never happen, since `activeValidators.length` is at most `numElectedValidators = round - initRounds < numberActiveValidators `. 

So I think this check is redundant, it should never be reached (could be tested) and should be removed.

Reply:
which condition are you referring to?
if it's L771 `if (activeValidators.length === this._moduleConfig.numberActiveValidators) {`, then this would happen in the loop because it's pushing into `activeValidators` until this condition is met?

Reply:
Now the code seems correct, can be resolved from my side.